source/Cute.Lib/Contentful/BulkActions/Actions/BulkActionBase.cs

```diff
@@ -344,15 +346,18 @@ private async Task<BulkActionResponse> SendBulkChangePublishRequestViaHttp(BulkA
 
     private async Task<BulkActionResponse> SendBulkActionStatusRequestViaHttp(string bulkActionResponseId, Action<BulkActionProgressEvent>? progressUpdater)
     {
-        var bulkEndpoint = new Uri($"https://api.contentful.com/spaces/{_contentfulConnection.Options.SpaceId}/environments/{_contentfulConnection.Options.Environment}/bulk_actions/actions/{bulkActionResponseId}");
+        var spaceId = (await _contentfulConnection.GetDefaultSpaceAsync()).SystemProperties.Id;
+        var environmentId = (await _contentfulConnection.GetDefaultEnvironmentAsync()).SystemProperties.Id;
+
+        var bulkEndpoint = new Uri($"https://api.contentful.com/spaces/{spaceId}/environments/{environmentId}/bulk_actions/actions/{bulkActionResponseId}");
```

Comment:
would it be consistent with the intended idea to move this inside of contentfulConnection?

Reply:
Yes and GraphQL- was next on the list